### PR TITLE
Add Pagination, Sorting, and Swagger to the existing controllers

### PR DESCRIPTION
**Pull request by DEz-1996**

### CODE REVIEW

File: src/main/java/online/book/store/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -1,15 +1,33 @@
...
+
+    @NotNull
+    @Size(min = MIN_STRING_LENGTH, max = MAX_STRING_LENGTH, message = STRING_SIZE_MSG)
     private String author;
+    @NotNull
+    @Size(min = MIN_STRING_LENGTH, max = MAX_STRING_LENGTH, message = STRING_SIZE_MSG)
+    private String title;
+
+    @NotNull
+    @Pattern(regexp = "\\d{10}|\\d{13}", message = ISBN_SIZE_MSG)
```
Comment (bhdnchui) [inline src/main/java/online/book/store/dto/CreateBookRequestDto.java]: Please, think about using org.hibernate.validator.constraints.ISBN annotation here.

File: src/main/java/online/book/store/exception/CustomGlobalExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getBindingResult().getAllErrors().stream()
+                .map(this::getErrorMessage)
+                .toList();
+        body.put("errors", errors);
+        return new ResponseEntity<>(body, headers, status);
+    }
+
```
Comment (bhdnchui) [inline src/main/java/online/book/store/exception/CustomGlobalExceptionHandler.java:37]: I think you have exceptions that you are not handling here, so let's add handles for them here.

File: src/main/java/online/book/store/repository/BookRepository.java [hunk 3]
```diff
@@ -2,8 +2,6 @@
 
 import online.book.store.model.Book;
 import org.springframework.data.jpa.repository.JpaRepository;
-import org.springframework.stereotype.Repository;
 
-@Repository
```
Comment (bhdnchui) [inline src/main/java/online/book/store/repository/BookRepository.java]: Why did you remove repository annotation?

**Review by bhdnchui — CHANGES_REQUESTED** [review]
Good job!
Just a few points to consider.

File: src/main/java/online/book/store/repository/BookRepository.java (hunk 3 above)
Comment (DEz-1996) [reply src/main/java/online/book/store/repository/BookRepository.java]: It was conflict with another PR
Fixed

**Review by vovasalyha — APPROVED** [review]
